ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql

```diff
@@ -0,0 +1,38 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+
+
+CREATE TABLE participant_users (
+    internal_id         SERIAL          PRIMARY KEY,
+    user_id             VARCHAR(256)            NOT NULL UNIQUE,
+    primary_party       VARCHAR(512),
+    -- The default is  microseconds since Unix epoch
+    created_at          BIGINT          NOT NULL DEFAULT (1000 * 1000 * EXTRACT(epoch FROM CURRENT_TIMESTAMP))::bigint
```

Comment:
fyi @nmarton-da

Reply:
The database might have a different time source than the participant. Could this lead to issues if you let the database assign a value here?

Reply:
There should be no consequences currently because `created_at` and `granted_at` are never used on the JVM side.
My understanding is their purpose is for auditing and having single source of timestamps (the DB) allows us to always compare them with each other. In contrast, having timestamps handled on JVM side could result in incomparable timestamps if there are multiple writing processes with out-of-sync clocks


fyi @meiersi-da

Reply:
I agree with @pbatko-da. It is better to have it in sql rather than in scala, where different participant copies on the same HA setup may have slightly differing clocks.

Reply:
> are never used on the JVM side. My understanding is their purpose is for auditing.

Thanks for the context, makes sense now.

Reply:
Although, if the purpose is for auditing only, would it not be more user friendly to use the `TIMESTAMP(Z)` data type?
```
created_at TIMESTAMPZ NOT NULL DEFAULT CURRENT_TIMESTAMP
```

Reply:
Trying to summarize pros/cons

Timestamps:
+ more user friendly,
+ simpler implementation in Oracle,
? chance of introducing timezone bugs,
Q: Is TIMESTAMPZ supported/behaves the same for all DBs? e.g. in case we wanted to implement some tooling making use of created_at and granted_at for all DBs.

Epoch micros:
+ already implemented (this + one more PR),
+ no timezone so little change for timezone bugs,
+ simple type: just a number (of microseconds ) have simpler semantics, easier to write tooling on top of it (less corner cases?),
- lengthy implementation in Oracle using before-insert triggers.

Reply:
Good question @rautenrieth-da and analysis @pbatko-da .

FWIW, I wouldn't be suprised if we ended up using these timestamps at some point on the JVM side, which is why I'd suggest to go with the uniform microseconds approach when storing them.

Reply:
For me it seems that having microseconds from the JVM:
* established/battle hardened pattern
* it is very simple and straight forward in terms of implementation (same implementation for all backends, as opposed to exotic complicated triggers for oracle)
* we already rely on proper time settings for the JVM, would that not be good enough already?

Reply:
Thanks @nmarton-da! I'll change to the JVM approach

Reply:
Probably
```
    val now = Instant.now()
    now.getEpochSecond * 1000 * 1000 + now.getNano / 1000
```